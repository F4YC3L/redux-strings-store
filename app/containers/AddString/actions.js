/*
 * Home Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  CHANGE_NEW_STRING,
  ADD_STRING_SUCCESS,
  ADD_STRING_ERROR,
  ADD_STRING,
} from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} newString The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_NEW_STRING
 */
export function changeNewString(newString) {
  return {
    type: CHANGE_NEW_STRING,
    newString,
  };
}

/**
 * add string, this action starts the request saga
 *
 * @return {object} An action object with a type of ADD_STRING
 */
export function addString() {
  return {
    type: ADD_STRING,
  };
}

/**
 * add string, this action starts the request saga
 *
 * @return {object} An action object with a type of ADD_STRING_SUCCESS
 */
export function stringAdded() {
  return {
    type: ADD_STRING_SUCCESS,
  };
}

/**
 * Dispatched when loading the data fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of ADD_STRING_ERROR passing the error
 */
export function stringAddingError(error) {
  return {
    type: ADD_STRING_ERROR,
    error,
  };
}
