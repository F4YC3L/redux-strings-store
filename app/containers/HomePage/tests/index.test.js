/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { HomePage, mapDispatchToProps } from '../index';
import { loadData } from '../../App/actions';
import configureStore from '../../../configureStore';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const loadDataFromApi = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <HomePage loading={false} error={false} data={[]} loadDataFromApi={loadDataFromApi} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the data on mount', () => {
    const loadDataFromApi = jest.fn();
    render(
      <Provider store={store}>
        <HomePage
          loadDataFromApi={loadDataFromApi}
        />
      </Provider>,
    );
    expect(loadDataFromApi).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {

    describe('loadDataFromApi', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.loadDataFromApi).toBeDefined();
      });

      it('should dispatch loadData when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.loadDataFromApi();
        expect(dispatch).toHaveBeenCalledWith(loadData());
      });
    });
  });
});
