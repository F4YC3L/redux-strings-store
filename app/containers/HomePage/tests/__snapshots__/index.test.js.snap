// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<article>
  <section
    class="Section-sc-1eulrsm-0 eTEsZl"
  >
    <div
      class="Wrapper-sc-1umgotm-0 laCwDT"
    >
      <ul
        class="Ul-mwnq6h-0 jcxXFy"
      >
        <li
          class="Wrapper-euo0oy-0 jLvxtc"
        >
          <div
            class="Item-sc-3y9mie-0 fKXSkT"
          >
            No data available to display, please add some strings!
          </div>
        </li>
      </ul>
    </div>
  </section>
</article>
`;
